src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -783,19 +798,43 @@ class StubGenerator: public StubCodeGenerator {
       __ prfm(use_stride ? Address(s, stride) : Address(s, prefetch), PLDL1KEEP);
 
     if (UseSIMDForMemoryOps) {
-      __ stpq(v0, v1, Address(d, 4 * unit));
-      __ ldpq(v0, v1, Address(s, 4 * unit));
-      __ stpq(v2, v3, Address(__ pre(d, 8 * unit)));
-      __ ldpq(v2, v3, Address(__ pre(s, 8 * unit)));
+      bs_asm->copy_store_at(_masm, decorators, type, 32,
+                            Address(d, 4 * unit), v0, v1,
+                            gct1, gct2, gct3, gcvt1, gcvt2, gcvt3);
+      bs_asm->copy_load_at(_masm, decorators, type, 32,
+                           v0, v1, Address(s, 4 * unit),
+                           gct1, gct2, gcvt1);
+      bs_asm->copy_store_at(_masm, decorators, type, 32,
+                            Address(__ pre(d, 8 * unit)), v2, v3,
+                            gct1, gct2, gct3, gcvt1, gcvt2, gcvt3);
+      bs_asm->copy_load_at(_masm, decorators, type, 32,
+                           v2, v3, Address(__ pre(s, 8 * unit)),
+                           gct1, gct2, gcvt1);
```

Comment:
All this extreme cut-and-paste manual unrolling is very hard to read, maintain, and review.
I wasn't going to say anything, because it's Erik, and what do I know! But I must push back here, this is too much.
Please consider these style changes.

Reply:
I updated the PR with a helper stack object encapsulating the choice of GC temp registers, types, decorators, etc, so that each line removes all the noise and becomes more readable. Do you like it? If yes, do you also want your proposed loop constructions in the new more compact form? I'm okay either way.

Reply:
The more compact form is a _huge_ improvement. I believe that using loops rather than hand-unrolling the code would make it even better!

Reply:
I think you are right that the loop form is easier to understand and maintain. But I think it makes it harder to convince myself that I didn't accidentally change the logic with this patch. I'm up for doing it as a follow-up RFE. Does that sound okay?